packages/node_modules/@ciscospark/internal-plugin-ediscovery/src/ediscovery.js

```diff
@@ -0,0 +1,520 @@
+/* eslint-disable prefer-template */
+/*!
+ * Copyright (c) 2015-2017 Cisco Systems, Inc. See LICENSE file.
+ */
+
+import {waitForValue, SparkPlugin} from '@ciscospark/spark-core';
+import reportRequest from './report-request';
+import ReportGenerator from './report-generator';
+
+/**
+ * @class Ediscovery is used by compliance officers to run compliance reports
+ *
+ */
+const Ediscovery = SparkPlugin.extend({
+  namespace: 'Ediscovery',
+
+  /**
+   * Creates a compliance report with a specific set of search parameters
+   *
+   * @param {String} name - A label to identify the report
+   * @param {String} description - A textual summary of the reports purpose
+   * @param {Array<String>} emails - A list of user emails relevant to the report
+   * @param {Array<String>} keywords - A list of search terms relevant to the report
+   * @param {Array<String>} spaceNames - A list of space names relevant to the report
+   * @param {Array<String>} spaceIds - A list of space ids relevant to the report
+   * @param {Object} range - Contains the start time and end time defining the search period
+   * @returns {Promise<ResponseEntity>} Http response containing the new report record
+   */
+  @waitForValue('@')
+  createReport(name, description, emails, keywords, spaceNames, spaceIds, range) {
+    if (!emails || !keywords || !spaceNames || !spaceIds || !range) {
+      throw Error('Invalid parameter(s)');
```

Comment:
I think you should add a `new` in front of `Error`. I haven't seen it this way before.

Reply:
Happy to go with convention, but I think this is the same as using _new_: http://es5.github.io/#x15.11.1
In general I've tried to avoid using _new_ where possible.